src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -242,5 +298,66 @@ private void DoBuild()
                 throw new Exception(message);
             }
         }
+
+        private void ThrowIfAssetSourcesHaveChanged()
+        {
+            if (!_dataDirectory.Exists)
+            {
+                return;
+            }
+
+            var dataDirectoryFiles = _dataDirectory.GetFiles("*", SearchOption.AllDirectories);
+
+            if (!dataDirectoryFiles.Any())
+            {
+                return;
+            }
+
+            var trackedFiles = _inventoryFiles.AllInventoryFiles.SelectMany(f => LoadInventory(f));
+
+            var assetFiles = GetFileList();
+
+            var untrackedFiles = assetFiles.Where(a => !trackedFiles.Any(t => t.FullName.Equals(a.FullName)));
```

Comment:
This is case sensitive and is going to give divergent behaviors on Windows vs *nix.